axelrod/tests/unit/test_match_generator.py

```diff
@@ -122,6 +122,29 @@ def test_build_single_match_params_with_prob_end_and_turns(self):
         self.assertGreater(len(match), 0)
         self.assertLessEqual(len(match), 10)
 
+    def test_build_single_match_params_with_fixed_length_unknown(self):
+        rr = axelrod.MatchGenerator(players=self.players,
+                                    game=test_game,
+                                    repetitions=test_repetitions,
+                                    turns=5,
+                                    match_attributes={"length": -1})
```

Comment:
For some reason I thought `-1` was the default already in place. It's not! Let me change it to `float('inf')` which is what is currently there. Good catch.